text/0000-api-evolution.md

```diff
@@ -0,0 +1,748 @@
+- Feature Name: not applicable
+- Start Date: 2015-05-04
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+
+This RFC proposes a comprehensive set of guidelines for which changes to
+*stable* APIs are considered breaking from a semver perspective, and which are
+not.  These guidelines are intended for both the standard library and for the
+crates.io ecosystem.
+
+This does *not* mean that the standard library should be completely free to make
+non-semver-breaking changes; there are sometimes still risks of ecosystem pain
+that need to be taken into account. Rather, this RFC makes explicit an initial
+set of changes that absolutely *cannot* be made without a semver bump.
+
+Along the way, it also discusses some interactions with potential language
+features that can help mitigate pain for non-breaking changes.
+
+The RFC covers only API issues; other issues related to language features,
```

Comment:
Redundant paragraph. Language, lints, type inference, CLI are not part of standard library (first paragraph), nor they are a part of crates.io ecosystem.